Libraries/Opc.Ua.Gds.Server.Common/ApplicationsNodeManager.cs

```diff
@@ -659,31 +663,87 @@ out nextRecordId
         {
             AuthorizationHelper.HasAuthenticatedSecureChannel(context);
 
-            //create CertificateValidator initialized with GDS CAs
-            var certificateValidator = new CertificateValidator();
-            var authorities = new CertificateTrustList() {
-                StorePath = m_globalDiscoveryServerConfiguration.AuthoritiesStorePath,
-                StoreType = CertificateStoreIdentifier.DetermineStoreType(m_globalDiscoveryServerConfiguration.AuthoritiesStorePath)
-            };
-            certificateValidator.Update(null, authorities, null);
-
-            //TODO return validityTime of Certificate once CertificateValidator supports it
+            //TODO return When the result expires and should be rechecked.
             validityTime = DateTime.MinValue;
 
-            using (var x509 = new X509Certificate2(certificate))
+            try
             {
-                try
+                //create chain to validate Certificate against it
+                var chain = new X509Chain();
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.Online;
+
+                using (var x509 = new X509Certificate2(certificate))
                 {
-                    certificateValidator.Validate(x509);
+                    if (chain.Build(x509))
+                    {
+                        return ServiceResult.Good;
+                    }
+                    else
+                    {
+                        certificateStatus = StatusCodes.Bad;
+                    }
                 }
-                catch (ServiceResultException se)
+            }
+            catch (CryptographicException)
+            {
+                certificateStatus = StatusCodes.Bad;
```

Comment:
should this set the StatusCode.BadCertificateRevoked or is just bad okay because it does not necessarily have to be a revocation error?
Or should we only care for revocation erros and ignore all other validation errors?